### PR TITLE
Clarify package upgrade and error if bad input

### PR DESCRIPTION
**Pull request by cawallin**
* Update docs to make it clear that you need to specify the RPM,
not the directory
* Error out if you specify a directory in which an RPM is,
rather than the actual location of the RPM

### CODE REVIEW

File: docs/presto-admin-commands.rst [hunk 1]
```diff
@@ -489,14 +489,14 @@ server upgrade
 **************
 ::
 
-    presto-admin server upgrade local_package_path [local_config_dir]
+    presto-admin server upgrade new-rpm [local_config_dir]
```
Comment (maciejgrzybek) [inline docs/presto-admin-commands.rst]: `path_to_new_rpm` maybe?

File: docs/presto-admin-commands.rst [hunk 2]
```diff
@@ -489,14 +489,14 @@ server upgrade
...
 
-This command upgrades the Presto RPM on all of the nodes in the cluster to the RPM specified
-by ``local_package_path``, preserving the existing configuration on the cluster. The existing
+This command upgrades the Presto RPM on all of the nodes in the cluster to the RPM
+``new-rpm``, preserving the existing configuration on the cluster. The existing
```
Comment (maciejgrzybek) [inline docs/presto-admin-commands.rst]: Maybe clarify if given path is relative to `presto-admin` binaries or it has to be absolute?

File: docs/presto-admin-commands.rst [hunk 3]
```diff
@@ -506,7 +506,7 @@ Example
 -------
 ::
 
-    sudo ./presto-admin server upgrade new-rpm.rpm /tmp/cluster-configuration
+    sudo ./presto-admin server upgrade /path/to/new-rpm.rpm /tmp/cluster-configuration
```
Comment (maciejgrzybek) [inline docs/presto-admin-commands.rst]: This should be corresponding to however we describe it here: https://github.com/prestodb/presto-admin/pull/182/files#r66441884.
If we describe it as absolute path, it's fine how's it now. If we are going to refer to it as a relative path, it should be relative in the example too.

File: docs/presto-admin-commands.rst (hunk 1 above)
Comment (cawallin) [inline docs/presto-admin-commands.rst]: I think the whole issue that John had was with the word path -- it made him think that he should specify the directory

File: tests/product/test_server_upgrade.py [hunk 4]
```diff
@@ -107,3 +107,28 @@ def test_rolling_upgrade(self):
...
+            'server upgrade ' + dir_on_cluster
+        )
+
+    def test_upgrade_works_with_symlink(self):
+        self.start_and_assert_started()
+
+        self.run_prestoadmin('configuration deploy')
+        for container in self.cluster.all_hosts():
+            self.real_installer.assert_installed(self, container)
+            self.assert_has_default_config(container)
```
Comment (maciejgrzybek) [inline tests/product/test_server_upgrade.py]: Shouldn't we assert some different configuration?
Even if upgrade overwrites configuration, it would overwrite it to default, so this test does not check if we're preserving configuration on upgrade.

Comment (maciejgrzybek) [inline tests/product/test_server_upgrade.py]: Maybe it should be a separate test though. E.g. `test_upgrade_preserves_configuration`.

File: docs/presto-admin-commands.rst (hunk 1 above)
Comment (maciejgrzybek) [inline docs/presto-admin-commands.rst]: What if you put `path/to/new/package.rpm`. Isn't it clear that you have to provide path with filename?

Comment (ebd2) [inline docs/presto-admin-commands.rst]: I second maciej's suggestion of path/to/new/package.rpm. It makes it clear that it it's a path to an rpm file.

File: docs/presto-admin-commands.rst (hunk 2 above)
Comment (ebd2) [inline docs/presto-admin-commands.rst]: Can't it be either?

File: prestoadmin/server.py [hunk 5]
```diff
@@ -183,8 +183,7 @@ def upgrade(local_package_path, local_config_dir=None, overwrite=False):
     Note that the configuration files in /etc/opt/prestoadmin are not updated
     during upgrade.
 
-    :param local_package_path - Absolute path to the presto rpm to be
-                                installed
+    :param new_rpm -            The new Presto RPM to be installed
```
Comment (ebd2) [inline prestoadmin/server.py]: How about "The path to the new Presto RPM to install"?

File: prestoadmin/package.py [hunk 6]
```diff
@@ -74,6 +74,10 @@ def deploy_action(local_path, rpm_action):
 
 
 def deploy(local_path=None):
+    if not os.path.isfile(local_path):
+        abort('RPM file not found at %s. Make sure to specify the filename'
+              'and not the directory name.' % local_path)
```
Comment (ebd2) [inline prestoadmin/package.py]: This seem like an awfully specific error message.

File: prestoadmin/package.py [hunk 7]
```diff
@@ -74,6 +74,10 @@ def deploy_action(local_path, rpm_action):
 
 
 def deploy(local_path=None):
+    if not os.path.isfile(local_path):
```
Comment (ebd2) [inline prestoadmin/package.py]: Should we verify that it's readable too? That seems like a common enough case that it shouldn't result in a stack trace for the user.

File: prestoadmin/server.py [hunk 8]
```diff
@@ -166,7 +166,7 @@ def uninstall():
 
 @task
 @requires_config(StandaloneConfig)
-def upgrade(local_package_path, local_config_dir=None, overwrite=False):
+def upgrade(new_rpm, local_config_dir=None, overwrite=False):
```
Comment (ebd2) [inline prestoadmin/server.py]: How about new_rpm_path?

File: docs/presto-admin-commands.rst (hunk 2 above)
Comment (maciejgrzybek) [inline docs/presto-admin-commands.rst]: Okay, let me rephrase it:
if we expect absolute path (e.g. because it's got to be available on all nodes in the same location or for whatever else reason), let's make it clear here.

File: tests/product/test_server_upgrade.py (hunk 4 above)
Comment (cawallin) [inline tests/product/test_server_upgrade.py]: good point. a little out of the scope of this PR, but I'll do it :)

File: prestoadmin/package.py (hunk 7 above)
Comment (cawallin) [inline prestoadmin/package.py]: You get a pretty helpful error message in put about this:
Fatal error: [hdp-kerb3-master] error: jre-8u77-linux-x64.rpm: open failed: Permission denied

That's probably more specific than what I would give

**Comment by cawallin** [comment]
Updated

File: prestoadmin/package.py (hunk 7 above)
Comment (ebd2) [inline prestoadmin/package.py]: Sounds reasonable; I'm OK leaving it as-is

File: prestoadmin/server.py [hunk 9]
```diff
@@ -183,8 +183,8 @@ def upgrade(local_package_path, local_config_dir=None, overwrite=False):
     Note that the configuration files in /etc/opt/prestoadmin are not updated
     during upgrade.
 
-    :param local_package_path - Absolute path to the presto rpm to be
-                                installed
+    :param new_rpm_path -       The path to the new Presto Presto RPM to
```
Comment (ebd2) [inline prestoadmin/server.py]: "Presto Presto"

File: tests/product/test_server_upgrade.py [hunk 10]
```diff
@@ -107,3 +107,68 @@ def test_rolling_upgrade(self):
...
+            self.assert_has_default_connector(container)
+
+        path_on_cluster = self.copy_upgrade_rpm_to_cluster()
+        symlink = '/opt/prestoadmin/link.rpm'
+        self.cluster.exec_cmd_on_host(self.cluster.master, 'ln -s %s %s'
+                                      % (path_on_cluster, symlink))
+        self.upgrade_and_assert_success(symlink)
+
+    def test_configuration_preserved_on_upgrade(self):
```
Comment (ebd2) [inline tests/product/test_server_upgrade.py]: Nice. Thanks!

File: tests/product/test_server_upgrade.py [hunk 11]
```diff
@@ -107,3 +107,68 @@ def test_rolling_upgrade(self):
...
+        symlink = '/opt/prestoadmin/link.rpm'
+        self.cluster.exec_cmd_on_host(self.cluster.master, 'ln -s %s %s'
+                                      % (path_on_cluster, symlink))
+
+        self.run_prestoadmin('server upgrade ' + path_on_cluster)
+        self.assert_dummy_properties(self.cluster.slaves[1])
+
+    def add_dummy_properties_to_host(self, host):
+        self.cluster.write_content_to_host(
+            'com.facebook.presto=INFO',
```
Comment (ebd2) [inline tests/product/test_server_upgrade.py]: Extract constants here and below?

**Comment by ebd2** [comment]
LGTM mod the typo and pulling out constants if you're inclined to do so.

**Comment by maciejgrzybek** [comment]
me to goods look
